utils/bitarray_utils.py

```diff
@@ -29,8 +29,8 @@ def uint_to_bitarray(x: int, bit_width=None) -> BitArray:
     converts an unsigned into to bits.
     if bit_width is provided then data is converted accordingly
     """
-    assert isinstance(x, int)
-    return int2ba(x, length=bit_width)
+    assert isinstance(x, (int, np.integer))
```

Comment:
Does this check specifically that x should be numpy integer?

Reply:
To Pulkit's question:
From https://www.w3schools.com/python/ref_func_isinstance.asp
> If the type parameter is a tuple, this function will return True if the object is one of the types in the tuple.

Reply:
@kedartatwawadi - are we using numpy integer types anywhere?

Reply:
resolved during our call